src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -24,406 +33,186 @@ class Kalman(BayesFiltSmooth):
-            Updated Normal RV (new filter estimate).
-        meas_rv : Normal
-            Measured random variable, as returned by the measurement model.
-        info : dict
-            Additional info. Contains at least the key `crosscov`,
-            which is the crosscov between input RV and measured RV.
-            The crosscov does not relate to the updated RV!
-        """
-        meas_rv, info = self.measure(time, randvar)
-        filt_rv = self.condition_state_on_measurement(
-            randvar, meas_rv, data, info["crosscov"]
-        )
-        return filt_rv, meas_rv, info
-
-    def smooth(self, filter_posterior, _intermediate_step=None):
-        """Apply Gaussian smoothing to the filtering outcome (i.e. a KalmanPosterior).
-
-        Parameters
-        ----------
-        filter_posterior : KalmanPosterior
-            Posterior distribution obtained after filtering
-        _intermediate_step :
-            Step-size to be taken by approximate transition methods.
-        Returns
-        -------
-        KalmanPosterior
-            Posterior distribution of the smoothed output
-        """
+    def smooth(self, filter_posterior):
         rv_list = self.smooth_list(
             filter_posterior,
             filter_posterior.locations,
-            _intermediate_step=_intermediate_step,
         )
         return KalmanPosterior(
             filter_posterior.locations, rv_list, self, with_smoothing=True
         )
 
-    def smooth_list(self, rv_list, locations, _intermediate_step=None):
-        """Apply smoothing to a list of RVs with desired final random variable.
-
-        Specification of a final RV is useful to compute joint samples from a KalmanPosterior object,
-        because in this case, the final RV is a Dirac (over a sample from the final Normal RV)
-        and not a Normal RV.
-
-        Parameters
-        ----------
-        rv_list : _RandomVariableList or array_like
-            List of random variables to be smoothed.
-        locations : array_like
-            Locations of the random variables in rv_list.
-        _intermediate_step :
-            Step-size to be taken by approximate transition methods.
-
-        Returns
-        -------
-        _RandomVariableList
-            List of smoothed random variables.
-        """
+    def smooth_list(self, rv_list, locations):
         final_rv = rv_list[-1]
         curr_rv = final_rv
         out_rvs = [curr_rv]
         for idx in reversed(range(1, len(locations))):
             unsmoothed_rv = rv_list[idx - 1]
-            curr_rv = self.smooth_step(
-                unsmoothed_rv,
-                curr_rv,
+
+            # Intermediate prediction
+            predicted_rv, info = self.predict(
+                rv=unsmoothed_rv,
                 start=locations[idx - 1],
                 stop=locations[idx],
-                _intermediate_step=_intermediate_step,
+            )
+            crosscov = info["crosscov"]
```

Comment:
these are necessary, because extracting the smoothing parts reduces down to only the update.

After #301 this will be easy to adress by caching pred_rvs as it should be.